src/sunmemory/sycl/sundials_sycl_memory.cpp

```diff
@@ -257,68 +250,63 @@ int SUNMemoryHelper_Dealloc_Sycl(SUNMemoryHelper helper, SUNMemory mem,
     {
       SUNDIALS_DEBUG_PRINT(
         "ERROR in SUNMemoryHelper_Dealloc_Sycl: unknown memory type\n");
-      return -1;
+      return SUN_ERR_CORRUPT;
```

Comment:
You'd used `SUN_ERR_OUTOFRANGE` in the CUDA version, but I think the most appropriate error code is
```suggestion
    return SUN_ERR_INCOMPATIBLE;
```